### PR TITLE
Support Resilience4J Configuration Properties

### PR DESCRIPTION
**Pull request by Ferioney**
included resilience4J spring boot 2 module
Fixes gh-61

### CODE REVIEW

**Review by ryanjbaxter — CHANGES_REQUESTED** [review]
This includes changes from #87, please separate the two.

File: README.adoc [hunk 1]
```diff
@@ -138,6 +138,39 @@ public Customizer<ReactiveResilience4JCircuitBreakerFactory> slowCustomizer() {
 ----
 ====
 
+
+==== Properties Configuration
+
+You can configure CircuitBreaker and TimeLimiter instances in Spring Boot’s `application.yml` config file. Property configuration has higher priority that Java `Customizer` configuration.
```
[the remark this answers is not in the document]
Comment (ryanjbaxter) [reply README.adoc:144]: "in Spring Boot’s  `application.yml`" -> "in your application's configuration properties"

**Review by ryanjbaxter — CHANGES_REQUESTED** [review]

File: spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/Resilience4JAutoConfiguration.java [hunk 2]
```diff
@@ -48,31 +44,10 @@
...
-	public Resilience4JCircuitBreakerFactory resilience4jCircuitBreakerFactory() {
-		Resilience4JCircuitBreakerFactory factory = new Resilience4JCircuitBreakerFactory();
+	public Resilience4JCircuitBreakerFactory resilience4jCircuitBreakerFactory(CircuitBreakerRegistry circuitBreakerRegistry,
+																			   TimeLimiterRegistry timeLimiterRegistry) {
+		Resilience4JCircuitBreakerFactory factory = new Resilience4JCircuitBreakerFactory(circuitBreakerRegistry, timeLimiterRegistry);
 		customizers.forEach(customizer -> customizer.customize(factory));
 		return factory;
 	}
-
-	@Configuration(proxyBeanMethods = false)
```
[the remark this answers is not in the document]
Comment (Ferioney) [reply spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/Resilience4JAutoConfiguration.java:57]: Reverted. 
Anyway, CircuitBreaker and TimeLimiter are automatically published on the Metrics, [Resilience4j Spring Boot 2](https://resilience4j.readme.io/docs/getting-started-3#metrics-endpoint)

File: README.adoc [hunk 3]
```diff
@@ -141,7 +141,7 @@ public Customizer<ReactiveResilience4JCircuitBreakerFactory> slowCustomizer() {
 
 ==== Properties Configuration
 
-You can configure CircuitBreaker and TimeLimiter instances in Spring Boot’s `application.yml` config file. Property configuration has higher priority that Java `Customizer` configuration.
+You can configure CircuitBreaker and TimeLimiter instances in your application's configuration properties config file. Property configuration has higher priority than Java `Customizer` configuration.
```
Comment (ryanjbaxter) [inline README.adoc:144]: remove "config"

File: spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/Resilience4JAutoConfiguration.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (ryanjbaxter) [reply spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/Resilience4JAutoConfiguration.java:57]: The `@Configuration` line still need to be added back

File: spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/Resilience4JCircuitBreaker.java [hunk 4]
```diff
@@ -61,9 +66,24 @@ public Resilience4JCircuitBreaker(String id,
...
+		this.timeLimiterRegistry = timeLimiterRegistry;
+		this.timeLimiterConfig = timeLimiterConfig;
+		this.executorService = executorService;
+		this.circuitBreakerCustomizer = circuitBreakerCustomizer;
+	}
+
 	@Override
 	public <T> T run(Supplier<T> toRun, Function<Throwable, T> fallback) {
-		TimeLimiter timeLimiter = TimeLimiter.of(timeLimiterConfig);
+		TimeLimiter timeLimiter = timeLimiterRegistry != null ? timeLimiterRegistry.timeLimiter(id, timeLimiterConfig)
```
Comment (ryanjbaxter) [inline spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/Resilience4JCircuitBreaker.java:85]: Seems to make sense to move this to the old constructor

File: spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/Resilience4JCircuitBreakerFactory.java [hunk 5]
```diff
@@ -59,14 +73,18 @@ public void configureDefault(
 		this.defaultConfiguration = defaultConfiguration;
 	}
 
+	@Deprecated
 	public void configureCircuitBreakerRegistry(CircuitBreakerRegistry registry) {
-		this.circuitBreakerRegistry = registry;
```
Comment (ryanjbaxter) [inline spring-cloud-circuitbreaker-resilience4j/src/main/java/org/springframework/cloud/circuitbreaker/resilience4j/Resilience4JCircuitBreakerFactory.java:63]: Add this line back

**Review by ryanjbaxter — CHANGES_REQUESTED** [review]

**Review by ryanjbaxter — APPROVED** [review]

**Pull request merged**